src/interface/meshblock_data.hpp

```diff
@@ -426,6 +426,19 @@ class MeshBlockData {
     return true;
   }
 
+  void SetAllVariablesToInitialized() {
+    std::for_each(varVector_.begin(), varVector_.end(),
+                  [](auto &sp_var) { sp_var->data.initialized = true; });
+  }
```

Comment:
The idea of the `initialized` flag is that a user can check it in a kernel to see if a variable has been freshly allocated with all zeros. In some use cases (none that are actually implemented right now), it would make sense to initialize with something other than zero. Users can register `package->InitNewlyAllocatedVars`, which will be called in the task `InitNewlyAllocatedVariables` before calls to `SetAllVariablesToInitialized` are made.

Reply:
Ok that makes sense. I guess my concern is that `SetAllVariablesToInitialized` could get called out of sync with the variable actually getting initialized, which could cause some subtle bug. I'm not sure what the solution is though.

Reply:
Yeah, I don't think there is a completely safe way to do this because `initialized` can't be changed on device.